addons/website/controllers/main.py

```diff
@@ -826,6 +830,29 @@ def google_maps_api_key(self):
     # Themes
     # ------------------------------------------------------
 
+    @http.route('/website/google_font_metadata', type='json', auth='user', website=True)
+    def google_font_metadata(self):
+        """ Avoid CORS by caching google fonts metadata on server """
+        Attachment = request.env['ir.attachment']
+        metadata = Attachment.search([
+            ('name', '=', 'googleFontMetadata'),
+            ('public', '=', True),
+        ])
+        yesterday = fields.Datetime.add(fields.Datetime.now(), days=-1)
+        if not metadata or metadata.write_date < yesterday:
+            json_text = requests.get('https://fonts.google.com/metadata/fonts', timeout=5).content
```

Comment:
Variable renamed and response check added.

Regarding the caching, this is not intended as a real proxy of Google's API, it's only to avoid a CORS issue. Given that the feature is likely to be used only when initially setting up a website, we decided to keep the result for one day so that the user can keep trying various fonts using the font selector without having our servers keep fetching the same 2MB of data over and over. At worst, some fonts in the list will not exist anymore, and new ones will only become available on the next day.